packages/ptokens-assets-evm/src/ptokens-evm-provider.ts

```diff
@@ -231,22 +236,25 @@ export class pTokensEvmProvider implements pTokensAssetProvider {
         return false
       }
     }, _pollingTime)
+    if (!log) throw new Error('Polling function stopped unexpectedly. No log found')
```

Comment:
This shouldn't be necessary. If we stick to use polling, the loop does not exit until true is returned, and there we return true only if there is a log!

Reply:
This is needed because of https://github.com/pnetwork-association/ptokens.js/blob/217b0755e47eb1b6466623bea6571e83edcf3c4b/packages/ptokens-assets-evm/src/ptokens-evm-provider.ts#L222
If not TS complains that log could be undefined.